lib/plugins/openaps.js

```diff
@@ -73,6 +73,14 @@ function init() {
     }
   };
 
+  openaps.findOfflineMarker = function findOfflineMarker(sbx) {
+    return _.findLast(sbx.data.treatments, function match(treatment) {
+      var eventTime = sbx.entryMills(treatment);
+      var eventEnd = treatment.duration ? eventTime + times.mins(treatment.duration).msecs : eventTime;
+      return eventTime <= sbx.time && treatment.eventType === 'OpenAPS Offline' && eventEnd >= sbx.time;
+    });
+  };
+
   openaps.updateVisualisation = function updateVisualisation (sbx) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (10)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1841765744/issues/source?bid=2452379&fileBranchId=3082539#l84)